modules/cold-extraction/ColdDataRetriever.py

```diff
@@ -15,109 +15,93 @@
-send_email = args['SendEmail']
-
-# Reads the system_json file.
-system_json = args['NifflerSystem']
-
-with open(system_json, 'r') as f:
-    niffler = json.load(f)
-
-# Get constants from system.json
-DCM4CHE_BIN = niffler['DCM4CHEBin']
-SRC_AET = niffler['SrcAet']
-QUERY_AET = niffler['QueryAet']
-DEST_AET = niffler['DestAet']
-NIGHTLY_ONLY = niffler['NightlyOnly']
-START_HOUR = niffler['StartHour']
-END_HOUR = niffler['EndHour']
-IS_EXTRACTION_NOT_RUNNING = True
-NIFFLER_ID = niffler['NifflerID']
-MAX_PROCESSES = niffler['MaxNifflerProcesses']
-
-SEPARATOR = ','
-
-accessions = []
-patients = []
-dates = []
-
-storescp_processes = 0
-niffler_processes = 0
-
-nifflerscp_str = "storescp.*{0}".format(QUERY_AET)
-qbniffler_str = 'ColdDataRetriever'
-
-niffler_log = 'niffler' + str(NIFFLER_ID) + '.log'
-
-logging.basicConfig(filename=niffler_log,level=logging.INFO)
-logging.getLogger('schedule').setLevel(logging.WARNING)
-
-# Variables to track progress between iterations.
-extracted_ones = list()
-
-# By default, assume that this is a fresh extraction.
-resume = False
-
-# All extracted files from the csv file are saved in a respective .pickle file.
-try:
-    with open(csv_file +'.pickle', 'rb') as f:
-        extracted_ones = pickle.load(f)
-        # Since we have successfully located a pickle file, it indicates that this is a resume.
-        resume = True
-except:
-    logging.info("No existing pickle file found. Therefore, initialized with empty value to track the progress to {0}.pickle.".format(csv_file))
-
-# record the start time
-t_start = time.time()
+def initialize_Values(valuesDict):
+    global storescp_processes, niffler_processes, nifflerscp_str, qbniffler_str
+    global storage_folder, file_path, csv_file, extraction_type, accession_index, patient_index, date_index, date_type, date_format, email, send_email, system_json
+    global DCM4CHE_BIN, SRC_AET, QUERY_AET, DEST_AET, NIGHTLY_ONLY, START_HOUR, END_HOUR, IS_EXTRACTION_NOT_RUNNING, NIFFLER_ID, MAX_PROCESSES, SEPARATOR
+    global accessions, patients, dates, niffler_log, resume, upload_folder
+
+    logs = [] # For showing logs on frontend if any error occurs
+
+    upload_folder = '../cold-extraction/csv/' # This is a check for frontend only,it won't conflict when program is run from terminal
+    if not os.path.exists(upload_folder):
+        os.makedirs(upload_folder)
+    storage_folder = valuesDict['storage_folder']
+    file_path = valuesDict['file_path']
+    csv_file = valuesDict['CsvFile']
+    extraction_type = valuesDict['extraction_type']
+    accession_index = int(valuesDict['accession_index'])
+    patient_index = int(valuesDict['patient_index'])
+    date_index = int(valuesDict['date_index'])
+    date_type = valuesDict['date_type']
+    date_format = valuesDict['date_format']
+    email = valuesDict['email']
+    send_email = bool(valuesDict['send_email'])
+    system_json = valuesDict['NifflerSystem']
+    
+    csv_file = upload_folder + csv_file
```

Comment:
This will confuse someone who develops the cold-extraction module. 

As above, this computation belongs to the frontend module. Please pass only the final csv_file value to cold-extraction and omit this line here.